server/schemas/schemaModel.js

```diff
@@ -13,31 +13,31 @@ bluebird.promisifyAll(Model);
 
 module.exports = {
   model: Model,
+  /*
+  createTemplate:
+    Accepts incoming request, response, and next (callback).
+    Attempts to find an existing schema with the provided name. If one does not exists,
+    a schema template is created from the request parameters.
 
+  */
   createTemplate: function(request, response, next) {
     var error;
-    var newSchema = request.body.schema;
-    if (!newSchema.name) {
-      error = {message: "Invalid schema name!"};
-      next(error, false);
-      return;
-    }
-    Model.findOneAsync({name: newSchema.name})
+    var newSchema = request.body;
+    var params = {
+      name: newSchema.name,
+      metaData: newSchema.metaData,
+      data: newSchema.data
+    };
+
+    Model.findOneAsync({name: params.name})
       .then(function(schema) {
-        //enable the following log statement for troubleshooting purposes
-        //console.debug("findOneAsync args:", arguments);
-        if (schema) {
-          error = {message: "Schema already exists"};
-          next(error, false);
-        } else {
-          return Model.createAsync({
-            name: newSchema.name,
-            metaData: newSchema.metaData,
-            data: newSchema.data
-          })
+        if (!schema) {
+          return Model.createAsync(params)
           .then(function(schema) {
             next(false, schema);
           });
+        } else {
```

Comment:
If there is not a match found by mongoose, it does not error, but just returns a falsy value for schema (null if I recall correctly). Thats why I explicitly handle this case.